File: Scenerio29.py
from pyspark import SparkConf
from pyspark import SparkContext
from pyspark import *
from pyspark.sql import SparkSession
from pyspark.sql import *
from pyspark.sql.types import *
from pyspark.sql.functions import *
from pyspark.sql.window import *

conf = SparkConf().setMaster("local[*]").setAppName("test")
sc = SparkContext(conf=conf)
sc.setLogLevel("ERROR")

spark = SparkSession.builder.getOrCreate()

data1 = [(1,), (2,), (3,)]

df1 = spark.createDataFrame(data1, ["col"])
df1.show()

data2 = [(1,), (2,), (3,), (4,), (5,)]

df2 = spark.createDataFrame(data2, ["col1"])
df2.show()

maxdf = df1.agg(max("col").alias("max"))
maxdf.show()

maxsalary = maxdf.select(col("max")).first()[0]

joindf = df1.join(df2, df1["col"] == df2["col1"], "outer").drop("col")
joindf.show()

finaldf = joindf.filter(col("col1") != maxsalary).withColumnRenamed("col1", "col").orderBy("col")
finaldf.show()

File: src/pack/Scenerio29.scala
package pack

import org.apache.spark.{SparkConf, SparkContext}
import org.apache.spark.sql.SparkSession
import org.apache.spark.sql._
import org.apache.spark.sql.types._
import org.apache.spark.sql.functions._
import org.apache.spark.sql.expressions._

object Scenerio29 {

  def main(args: Array[String]): Unit = {
    val conf = new SparkConf().setMaster("local[*]").setAppName("Scenerio27")
    val sc = new SparkContext(conf)
    sc.setLogLevel("ERROR")
    val spark = SparkSession.builder().getOrCreate()
    import spark.implicits._

    val df1 = Seq((1), (2), (3)).toDF("col")
    df1.show()

    val df2 = Seq((1), (2), (3), (4), (5)).toDF("col1")
    df2.show()

    val maxdf = df1.agg(max("col").as("max"))
    maxdf.show()

    val maxsalary = maxdf.select(col("max")).first().getInt(0)

    val joindf = df1.join(df2, df1("col") === df2("col1"), "outer").drop("col")
    joindf.show()

    val finaldf = joindf.filter(col("col1") =!= maxsalary).withColumnRenamed("col1", "col")
    finaldf.show()

  }
}
